src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -428,9 +428,16 @@ class ArrayLiteralSegment(BaseSegment):
     """An array literal segment."""
 
     type = "array_literal_type"
-    match_grammar = Bracketed(
-        Delimited(Ref("ExpressionSegment")),
-        bracket_type="square",
+    match_grammar = OneOf(
+        Bracketed(
+            Delimited(Ref("ExpressionSegment")),
+            bracket_type="square",
+        ),
+        Sequence(
```

Comment:
I was surprised to notice that `Bracketed` would not parse an empty list (`[]`). I added this to make it work, but I wonder if `Bracketed` **should** support this, either by default or as a new keyword option.

Reply:
Does making `Delimited(Ref("ExpressionSegment")),` `Delimited(Ref("ExpressionSegment"), optional=True),` solve this?

Reply:
Ha, of course! That's a much better way to do it. I created a mental "trap" for myself, thinking this should be a feature of `Bracketed` instead. 🙄